src/Symfony/Component/Messenger/Transport/Doctrine/Connection.php

```diff
@@ -52,12 +53,14 @@ class Connection
     private $configuration = [];
     private $driverConnection;
     private $schemaSynchronizer;
+    private $schemaConfig;
 
-    public function __construct(array $configuration, DBALConnection $driverConnection, SchemaSynchronizer $schemaSynchronizer = null)
+    public function __construct(array $configuration, DBALConnection $driverConnection, SchemaSynchronizer $schemaSynchronizer = null, SchemaConfig $schemaConfig = null)
     {
         $this->configuration = array_replace_recursive(self::DEFAULT_OPTIONS, $configuration);
         $this->driverConnection = $driverConnection;
         $this->schemaSynchronizer = $schemaSynchronizer ?? new SingleDatabaseSynchronizer($this->driverConnection);
+        $this->schemaConfig = $schemaConfig ?? ($this->driverConnection->getSchemaManager() ? $this->driverConnection->getSchemaManager()->createSchemaConfig() : null);
```

Comment:
`getSchemaManager` does not return nullable value. so the check you added is useless.

Reply:
Fixed. I also had to adapt the tests to add mocks